share/transition_scripts/6.6.0/openldap2ds389.rb

```diff
@@ -0,0 +1,179 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+#
+#  This script is used to convert data from a SIMP openldap server into a
+#  an ldif that can be imported into a SIMP ds389 server that has the same base dn.
+#
+#    This is a list of what it changes on the LDIF entries:
+#
+#    - Moves entries from ou=Group,<basedn> to ou=Groups,<basedn>
+#      dsidm, a tool used to maintain entries in ds389, is opinionated at
+#      this time on where groups and users should be.  It expects them
+#      to be in  ou=Groups and ou=People.
+#    - Removes attributes that are not compatible with ds389 (see
+#      @delete_keys for a full list)
+#    - Converts SSH  attributes to the format used by ds389.
+#    - Adds objectClass attributes needed by dsidm.
+#    - Converts lockout settings.
+#
+require 'fileutils'
+require 'net-ldap'
+require 'optparse'
+
+def convert_group(dn,attr)
+  newattr =  {}
+  newdn = dn.gsub(@oldgroupdn,@newgroupdn)
+  attr.each do |a,v|
+    case a
+    when :objectclass
+      newv = [ 'groupOfNames', 'nsMemberOf', 'posixGroup']
+      v.each do |nv|
+        newv << nv
+      end
+      newattr[:objectclass] = newv.uniq
+    else
+      newattr[a] = v unless @delete_keys.member?(a)
+    end
+  end
+  @ldifs_389[newdn] = newattr
```

Comment:
From the calling code, it's sort of a surprise that this happens to @ldifs_389 here.  

As a slight tweak: if the function simply returned the converted key/value pair, like

```ruby
{newdn => newattr}
```

then the calling code could take care of merging it into `@ldifs_389`:

```ruby
@ldifs_389.merge! convert_group(<args>)
```
and  `@ldifs_389` could just be a normally-scoped variable, i.e.,

```
ldifs_389.merge! convert_group(<args>)
```

That would make everything easier to keep track of and keep the intent of the function a bit clearer